standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -101,7 +101,7 @@ public void setConf(Configuration conf) {
   @Override
   public void alterTable(RawStore msdb, Warehouse wh, String catName, String dbname,
       String name, Table newt, EnvironmentContext environmentContext,
-      IHMSHandler handler, String writeIdList)
+      IHMSHandler handler, String writeIdList, boolean isTruncateOp)
```

Comment:
instead of adding a new `isTruncateOp`, can we put it into the environmentContext, like` environmentContext.putToProperties("__isTruncateOp__", "true");`